tasks.py

```diff
@@ -1355,6 +1355,7 @@ def test(
         'dev': 'Set up development environment at the end',
         'validate_files': 'Validate media files are correctly copied',
         'use_ssh': 'Use SSH protocol for cloning the demo dataset (requires SSH key)',
+        'branch': 'Specify branch of demo-dataset to clone (default = main)',
```

Comment:
Do we need this in this PR? I can see the value in general

Reply:
It's a handy way of testing against a specific branch, which is required for this PR (until it's merged). I think it's a reasonably handy feature to leave in general